nima/common/tls/src/main/java/io/helidon/nima/common/tls/TLSUtils.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.nima.common.tls;
+
+import java.security.Principal;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+
+/**
+ * Utility class for TLS.
+ */
+public class TLSUtils {
+
+    private TLSUtils() {
+    }
+
+    /**
+     * Parse the Common Name (CN) from certificates.
+     *
+     * @param certificates certificates
+     * @return  Common Name value
+     */
+    public static Optional<String> parseCN(Certificate[] certificates) {
```

Comment:
The method name should be `parseCn` instead

Reply:
Also as this accepts an array, the Javadoc is not valid. It should say it parses CN from the first certificate (if any present) or similar instead